src/main/java/gregtech/common/covers/detector/CoverDetectorEnergyAdvanced.java

```diff
@@ -66,166 +61,222 @@ public EnumActionResult onScrewdriverClick(EntityPlayer playerIn, EnumHand hand,
+                "cover.advanced_energy_detector.normal", "cover.advanced_energy_detector.inverted")
+                .setTooltipHoverString("cover.advanced_energy_detector.invert_tooltip")
         );
 
-        return ModularUI.builder(GuiTextures.BACKGROUND, 176, 108)
+        return ModularUI.builder(GuiTextures.BACKGROUND, 176 + (3 * SIZE), 108 + (SIZE))
                 .widget(group)
+                .widget(widgetsToUpdate) // add synced widgets
                 .build(this, player);
     }
 
+    private WidgetGroup constructWidgetsToUpdate() {
+        WidgetGroup sync = new WidgetGroup();
+        sync.addWidget(new TextFieldWidget2(76, 5 + (SIZE + PADDING), 8 * SIZE, SIZE, this::getMinValue, this::setMinValue)
+                .setMaxLength(19)
+                .setAllowedChars(TextFieldWidget2.NATURAL_NUMS)
+                .setPostFix(this.getPostFix()));
+        sync.addWidget(new TextFieldWidget2(76, 5 + 2 * (SIZE + PADDING), 8 * SIZE, SIZE, this::getMaxValue, this::setMaxValue)
+                .setMaxLength(19)
+                .setAllowedChars(TextFieldWidget2.NATURAL_NUMS)
+                .setPostFix(this.getPostFix()));
+        return sync;
+    }
+
     private String getMinValue() {
-        return String.valueOf(minEU);
+        return String.valueOf(minValue);
     }
 
     private String getMaxValue() {
-        return String.valueOf(maxEU);
+        return String.valueOf(maxValue);
     }
 
     private void setMinValue(String val){
-        try {
-            long c = Long.parseLong(val);
-            this.minEU = Math.min(maxEU - 1, c);
-        } catch (NumberFormatException e) {
-            GTLog.logger.warn(e);
-            this.minEU = Math.max(maxEU - 1, 0);
+        long parsedValue;
+
+        if (usePercent) {
+            parsedValue = GTUtility.tryParseLong(val, DEFAULT_MIN_PERCENT);
+        } else {
+            parsedValue = GTUtility.tryParseLong(val, DEFAULT_MIN_EU);
         }
+
+        this.minValue = Math.min(this.maxValue - 1, Math.max(0, parsedValue));
     }
 
     private void setMaxValue(String val){
-        try {
-            long c = Long.parseLong(val);
-            maxEU = Math.max(minEU + 1, c);
-        } catch (NumberFormatException e) {
-            GTLog.logger.warn(e);
-            this.maxEU = Math.max(minEU + 1, 2048);
+        long parsedValue;
+        long maxUpperLimit;
+
+        if (usePercent) {
+            parsedValue = GTUtility.tryParseLong(val, DEFAULT_MAX_PERCENT);
+            maxUpperLimit = 100;
+        } else {
+            parsedValue = GTUtility.tryParseLong(val, DEFAULT_MAX_EU);
+            maxUpperLimit = Long.MAX_VALUE;
         }
+
+        this.maxValue = Math.max(this.minValue + 1, Math.min(parsedValue, maxUpperLimit));
     }
 
     private boolean isInverted(){
-        return this.inverted;
+        return this.isInverted;
     }
```

Comment:
Optional (as I will go touch all of cover detectors later): This should be in `CoverDetectorEnergy` so it is consistent with other Advanced versions